catalog/ui/catalog-ui-search/src/main/webapp/component/workspace-lists/workspace-lists.view.js

```diff
@@ -83,9 +83,9 @@ module.exports = Marionette.LayoutView.extend({
         this.listenTo(this.listSelect.currentView.model, 'change:value', this.updateResultsList);
         this.listenTo(this.listSelect.currentView.model, 'change:value', this.handleSelection);
         this.listenTo(this.listSelect.currentView.model, 'change:value', this.handleEmptyList);
-        this.listenTo(this.model, 'add remove update', this.handleEmptyLists);
-        this.listenTo(this.model, 'add remove update', this.handleSelection);
-        this.listenTo(this.model, 'change:list.bookmarks', this.handleEmptyList);
+        this.listenTo(this.model, 'remove update', this.handleEmptyLists);
+        this.listenTo(this.model, 'remove update', this.handleSelection);
+        this.listenTo(this.model, 'change:list.bookmarks', this.handleBookmarksChange);
```

Comment:
I was able to refactor this file to be a less volatile / painful to work with by making the `updateResultsList` function idempotent.

here's the commit: https://github.com/codice/ddf/commit/27326951f9f0de05dc041890e04588980268dbc9

Reply:
thanks, I applied the patch